### ISSUE TITLE
MaskRcnn-resnet101 with cuda can not run

### ISSUE THREAD

**Issue by sugger23** [opencv/opencv#17716]
##### System information (version)
- OpenCV => 4.3
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017
-cuda 10.0
-->

maskrcnn-inceptionV2 can run both on CPU and GPU  sucessufully
but  when I try to use maskrcnn-resnet run with opencv, it can only run on CPU sucessufully,  GPU  is not supported ,it can not run "net.forward()"

**Comment by YashasSamaga** [comment]
Please share the model.

What happens when you try to run it on GPU? Wrong results? Error?

**Comment by sugger23** [comment]
> Please share the model.
> 
> What happens when you try to run it on GPU? Wrong results? Error?

@YashasSamaga  I add tyr..catch   on  "net.forward" ,    catch(std::exception e)   get the error info:  Unknown exception

my GPU is 1080TI

**Comment by sugger23** [comment]
> Please share the model.
> 
> What happens when you try to run it on GPU? Wrong results? Error?

I add try..catch on "net.forward" , catch(std::exception e) get the error info: Unknown exception

my GPU is 1080TI

**Comment by sugger23** [comment]
> Please share the model.
> 
> What happens when you try to run it on GPU? Wrong results? Error?

@YashasSamaga  I have send the model file to your E-MAIL,

**Comment by YashasSamaga** [comment]
@sugger23 The model executes without errors on my device.

What exactly was the error message you got?

**Comment by YashasSamaga** [comment]
I have got out-of-memory on RTX 2080 Ti with 11GB RAM with FP32 target. Can you please confirm if you are facing out-of-memory error?

**Comment by sugger23** [comment]
> I have got out-of-memory on RTX 2080 Ti with 11GB RAM with FP32 target. Can you please confirm if you are facing out-of-memory error?

@YashasSamaga I watch the GPU-Z when the program runs, the max memory used is 2.5GB,   my gpu is 1080TI,
I got the opencv4.3 two month ago ,maybe it have been soloved on the current github, I will try the  Newest github code soon

**Comment by sugger23** [comment]
> I have got out-of-memory on RTX 2080 Ti with 11GB RAM with FP32 target. Can you please confirm if you are facing out-of-memory error?

@YashasSamaga tody 
    Because my gpu is 1080TI，it don't support fp16，so I do the inference by f32,  I watch the GPU-Z when the program runs, the max memory used is 2.5GB.
    From your idea, I set FP16, then the program can run correctly, but it's too slow, even slower than CPU.(CPU time-9S ,  1080TI-FP16 time-11S),and it used the gpu memory about 1.8G.

   Then I get a 2060super GPU, it was faster, about 130MS,  gpu memory about 2.1G.    But F32 mode is the same as 1080TI, Unknown exception, gpu memory about 2.5G
   
   In summary，I think, it can run  correctly on the gpu which support FP16 ,such as 2080  2060 and so on. but can't run correctly on the GPU such as 1080 1060

    At last， Thank you very much,!

**Comment by YashasSamaga** [comment]
@sugger23 Can you share an exact copy of the error message? I think there should be some more message after Unknown Exception. The exception will have the line number and file name along with it. It will help isolate the problem.

**Comment by dkurt** [comment]
Closing due no respond from author
